includes/util.php

```diff
@@ -495,49 +495,116 @@ function newspack_get_countries() {
+ *
+ * @param  string $hex Hexidecimal value of the color to adjust.
+ *
+ * @return string Either black or white hexidecimal values.
+ *
+ * @ref https://stackoverflow.com/questions/1331591/given-a-background-color-black-or-white-text
+ */
+function newspack_get_color_contrast( $hex ) {
+	// hex RGB.
+	$r1 = hexdec( substr( $hex, 1, 2 ) );
+	$g1 = hexdec( substr( $hex, 3, 2 ) );
+	$b1 = hexdec( substr( $hex, 5, 2 ) );
+	// Black RGB.
+	$black_color    = '#000';
+	$r2_black_color = hexdec( substr( $black_color, 1, 2 ) );
+	$g2_black_color = hexdec( substr( $black_color, 3, 2 ) );
+	$b2_black_color = hexdec( substr( $black_color, 5, 2 ) );
+	// Calc contrast ratio.
+	$l1             = 0.2126 * pow( $r1 / 255, 2.2 ) +
+		0.7152 * pow( $g1 / 255, 2.2 ) +
+		0.0722 * pow( $b1 / 255, 2.2 );
+	$l2             = 0.2126 * pow( $r2_black_color / 255, 2.2 ) +
+		0.7152 * pow( $g2_black_color / 255, 2.2 ) +
+		0.0722 * pow( $b2_black_color / 255, 2.2 );
+	$contrast_ratio = 0;
+	if ( $l1 > $l2 ) {
+		$contrast_ratio = (int) ( ( $l1 + 0.05 ) / ( $l2 + 0.05 ) );
+	} else {
+		$contrast_ratio = (int) ( ( $l2 + 0.05 ) / ( $l1 + 0.05 ) );
+	}
+	if ( $contrast_ratio > 5 ) {
+		// If contrast is more than 5, return black color.
+		return 'black';
+	} else {
+		// if not, return white color.
+		return 'white';
+	}
+}
+
+/**
+ * Get theme primary and secondary colors or defaults if none present.
+ *
+ * @return array An array containing primary and secondary colors.
+ */
+function newspack_get_theme_colors() {
+	$default_primary_color   = function_exists( 'newspack_get_primary_color' ) ? newspack_get_primary_color() : '#3366ff';
+	$default_secondary_color = function_exists( 'newspack_get_secondary_color' ) ? newspack_get_secondary_color() : '#f0f0f0';
+	$primary_color           = get_theme_mod( 'primary_color_hex', $default_primary_color );
+	$secondary_color         = get_theme_mod( 'secondary_color_hex', $default_secondary_color );
+
+	return [
+		'primary_color'        => $primary_color,
+		'secondary_color'      => $secondary_color,
+		'primary_text_color'   => newspack_get_color_contrast( $primary_color ),
+		'secondary_text_color' => newspack_get_color_contrast( $secondary_color ),
+	];
+}
+
 /**
  * Get block and html markup for social media services.
  *
+ * @param string $color The color to use for the social media icons.
+ *
  * @return array An array containing block and html markup for social media services.
  */
-function newspack_get_social_markup() {
+function newspack_get_social_markup( $color = 'white' ) {
 	$cm          = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'wordpress_seo' );
 	$social_urls = [];
+	$markup      = [
+		'block_markup' => '',
+		'html_markup'  => '',
+	];
+
 	if ( ! is_wp_error( $cm ) ) {
 		$social_urls = [
+			// TODO: Add more social media services here.
```

Comment:
Why not add them now? From what I see [here](https://github.com/Automattic/newspack-plugin/blob/8b5d5c4c8b7372dd33603988142097435997bf2a/includes/class-profile.php#L36-L69), only LinkedIn and Pinterest are missing.

Reply:
Fair point! I've added linkedin in https://github.com/Automattic/newspack-plugin/pull/3033/commits/ff88d4b851c20019c59edacd517f9237a95f0f53

It looks like newsletters doesn't support pinterest though so I've omitted that one for the time being:
https://github.com/Automattic/newspack-newsletters/blob/6711238e09a20dd215833207810e9e341cd54a3f/includes/class-newspack-newsletters-renderer.php#L164-L173

I've added this to the list of small changes needed in newsletters and will tackle this altogether in a follow-up PR: 0-as-1205668937699542/1207101310249278/f